cmd/thanos/downsample.go

```diff
@@ -419,7 +420,7 @@ func processDownsampling(
 
 	err = block.Upload(ctx, logger, bkt, resdir, hashFunc)
 	if err != nil {
-		return errors.Wrapf(err, "upload downsampled block %s", id)
+		return compact.NewRetryError(errors.Wrapf(err, "upload downsampled block %s", id))
```

Comment:
From scanning the code, I dont think this is consumed somewhere right now, right? I dont think this will lead to retries currently!

Reply:
As I understand it
`processDownsampling` returns error to
`downsampleBucket` which returns error to `compactMainFn` in `cmd/thanos/compact.go`
then in `cmd/thanos/compact.go` there's a `if compact.IsRetryError(err) {` check
which should trigger

Reply:
assuming we run compactor with `--wait` flag

Reply:
Maybe it's better to retry right here instead of returning an error. This way the compactor will not have to go through the whole cycle again and downsample the block from the beginning.

Reply:
You are right, my bad!

Reply:
@xBazilio sounds good, we can keep things consistent for now. Nothing is set in stone anyway, so we can improve if needed.